From ba0c6c70045b0aa355ed421e4fdf65511f80eb04 Mon Sep 17 00:00:00 2001
From: raosrini <srinidhi.rao@intel.com>
Date: Tue, 14 Apr 2015 15:47:45 +0530
Subject: [PATCH] dc_ti_cc: Adapt to new coulomb counter interfaces

some new sysfs interfaces to support unique features of
a coulomb counter have been added in the S/W  FG Driver.
Hence the $Cove TI Coulomb Counter driver has to export
the APIs for the unique features that it supports. If the
Coulomb Counter does not support a feature for e.g. like
Interrupt, then it can ignore that feature by assigning
a dummy API.

Change-Id: I0d8cac1b6ce4016ca647a7cb6c52b87d622a6256
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1400
Signed-off-by: raosrini <srinidhi.rao@intel.com>
---
 drivers/platform/x86/dc_ti_cc.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/drivers/platform/x86/dc_ti_cc.c b/drivers/platform/x86/dc_ti_cc.c
index 24515c1..fb3fea0 100644
--- a/drivers/platform/x86/dc_ti_cc.c
+++ b/drivers/platform/x86/dc_ti_cc.c
@@ -692,6 +692,17 @@ static int dc_ti_cc_calibrate(void)
 	return 0;
 }
 
+static int unsupported_read(int *val)
+{
+	*val = 0;
+	return *val;
+}
+
+static int unsupported_write(int val)
+{
+	return 0;
+}
+
 static struct intel_fg_input fg_input = {
 	.get_batt_params = &dc_ti_cc_get_batt_params,
 	.get_v_ocv = &dc_ti_cc_get_vocv,
@@ -701,6 +712,17 @@ static struct intel_fg_input fg_input = {
 	.get_i_avg = &dc_ti_cc_get_iavg,
 	.get_delta_q = &dc_ti_cc_get_deltaq,
 	.calibrate_cc = &dc_ti_cc_calibrate,
+
+	.get_up_cc = &unsupported_read,
+	.get_down_cc = &unsupported_read,
+	.get_acc_err = &unsupported_read,
+	.get_delta_thr = &unsupported_read,
+	.get_long_avg = &unsupported_read,
+	.get_long_avg_ocv = &unsupported_read,
+	.get_ocv_accuracy = &unsupported_read,
+	.reset_acc_err = &unsupported_write,
+	.set_delta_thr = &unsupported_write,
+	.clr_latched_ibat_avg = &unsupported_write,
 };
 
 static void dc_ti_update_boot_ocv(struct dc_ti_cc_info *info)
-- 
1.9.1

